### ISSUE TITLE
result is much worse than ohem

### ISSUE THREAD

**Issue by twmht** [libuyu/GHM_Detection#33]
Hi,

I have trained a face detector based on ohem loss (https://github.com/biubug6/Pytorch_Retinaface/blob/master/layers/modules/multibox_loss.py#L103)

and now I have replaced the regression loss and classification loss with ghm-c and ghm-r.

However, it seems that ohem is much better.

Here is a example,

from ohem,

![test_ohem](https://user-images.githubusercontent.com/1567200/77519295-3c212900-6eba-11ea-93f0-8389d321b5e7.jpg)

the probabilities of the foreground are

```
0.997
0.98
0.98
0.97
0.81
```

![test_ghm](https://user-images.githubusercontent.com/1567200/77519521-9f12c000-6eba-11ea-98af-54de5e963688.jpg)

the foreground probabilities are from [0.5,0.6], it's very low. and background probabilities seems not decrease.

the training loss seems fine.

![Selection_014](https://user-images.githubusercontent.com/1567200/77519725-f44ed180-6eba-11ea-8094-d7c38a6e0c3e.png)

by the way,  i have trained 30 epochs, however, the probabilities of foreground object is much lower when training more epochs . Even the training loss from last epoch is much lower than early epoch. It seems that this is overfitting?

here is the early epoch (epoch-9) from ghm.

![test_ghm_9](https://user-images.githubusercontent.com/1567200/77520120-aa1a2000-6ebb-11ea-889c-14516d0e140a.jpg)

the probabilities of foreground object is much higher than last epoch, it's about [0.6,0.7]. but still there are many background.

I use the confidence threshold as 0.5, and I did the initialization trick as the focal loss.

not sure how to look into this problem and tune the parameters (I use the same ghm parameters as yours)

thank you.
